File: backend/app/schemas/analytics.py
from __future__ import annotations

from pydantic import BaseModel
from typing import List


class RfiAgingPoint(BaseModel):
    status: str
    count: int
    avg_age_days: float


class TopIssuePoint(BaseModel):
    issue: str
    count: int


class TradeRiskPoint(BaseModel):
    trade: str
    open_rfis: int
    risk_score: float


class AnalyticsResponse(BaseModel):
    rfi_aging: List[RfiAgingPoint]
    top_issues: List[TopIssuePoint]
    trade_risk: List[TradeRiskPoint]

File: frontend/src/api/analytics.ts
import { api } from './client'

export type Analytics = {
  rfi_aging: { status: string; count: number; avg_age_days: number }[]
  top_issues: { issue: string; count: number }[]
  trade_risk: { trade: string; open_rfis: number; risk_score: number }[]
}

export async function getAnalytics(projectId: string): Promise<Analytics> {
  const res = await api.get(`/projects/${projectId}/analytics`)
  return res.data
}
